gguf-py/gguf/gguf_writer.py

```diff
@@ -845,7 +847,14 @@ def _pack_val(self, val: Any, vtype: GGUFValueType, add_vtype: bool) -> bytes:
             encoded_val = val.encode("utf-8") if isinstance(val, str) else val
             kv_data += self._pack("Q", len(encoded_val))
             kv_data += encoded_val
-        elif vtype == GGUFValueType.ARRAY and isinstance(val, Sequence) and val:
+        elif vtype == GGUFValueType.ARRAY:
+
+            if not isinstance(val, Sequence):
+                raise ValueError(f"Invalid GGUF metadata array, expecting sequence")
```

Comment:
```suggestion
                raise ValueError("Invalid GGUF metadata array, expecting sequence")
```

Otherwise flake8 complains